client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyResponse.java

```diff
@@ -64,19 +74,12 @@ public ByteBuf apply(List<ByteBuf> l) {
     public Single<byte[]> bodyAsByteArrayAsync() {
         return collectContent().map(new Function<ByteBuf, byte[]>() {
             @Override
-            public byte[] apply(ByteBuf byteBuf) {
-                return toByteArray(byteBuf);
+            public byte[] apply(ByteBuf byteBuf) throws Exception {
+                return byteBuf.array();
```

Comment:
Doesn't this function expose the internal array of the ByteBuf? I know that you've done a lot to avoid copying bits, but I just want to make sure you're confident that you want to expose the internal array of a ByteBuf.

Reply:
It does. The reason that's ok is that the ByteBuf received by this method comes from `Unpooled.buffer` on line 59. Actually, I would expect a buffer from the pool to throw because there's only direct memory backing it and not an array.

It might be worth playing with how we aggregate the ByteBuf content for a more efficient conversion to byte[] or String (probably String is a more important case) but for now it's sufficient to just make sure we don't use operators like toList() which risk leaks.